golang/cosmos/x/swingset/keeper/keeper.go

```diff
@@ -76,6 +78,61 @@ func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
 	k.paramSpace.SetParamSet(ctx, &params)
 }
 
+func (k Keeper) GetBeansPerUnit(ctx sdk.Context) map[string]sdk.Uint {
+	params := k.GetParams(ctx)
+	beansPerUnit := make(map[string]sdk.Uint, len(params.BeansPerUnit))
+	for _, bpu := range params.BeansPerUnit {
+		beansPerUnit[bpu.Key] = bpu.Beans
+	}
+	return beansPerUnit
+}
+
+func (k Keeper) GetBeansOwing(ctx sdk.Context, addr sdk.AccAddress) sdk.Uint {
+	path := "beansOwing." + addr.String()
```

Comment:
Consider an addr-to-path-string function in common with `SetBeansOwing`, though if you think it's not worth the effort I'd probably agree.